bitbucket-pipelines.yml

```diff
@@ -0,0 +1,91 @@
+# Define the Docker image to use for all steps
+image: 
+  name: aaronmedinaaligent/shopify-cli-base:latest
+  username: $DOCKER_HUB_USERNAME
+  password: $DOCKER_HUB_PASSWORD
+
+definitions:
+  steps:
+    - step: &theme-check
+        name: Theme Check
+        services:
+          - docker
+        caches:
+          - docker
+        script:
+          - echo "STARTING THEME CHECK..."
+          - echo "Verifying Ruby, Node.js, and Shopify CLI installations..."
+          - ruby -v      
+          - node -v     
+          - npm -v
+          - shopify version  
+          - ls -al        
+          - echo "Checking if /theme directory exists..."
+          - if [ ! -d "./theme" ]; then echo "/theme directory does not exist."; exit 1; fi
+          - echo "Listing files in the /theme directory..."
+          - find theme -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
+          - echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+          - shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+
+    - step: &deploy
+        name: Deploy Theme
+        services:
+          - docker
+        caches:
+          - docker
+        script:
+          - echo "Deploying theme..."
+          - ruby -v
```

Comment:
Are these checks made redundant if the base image caters for them?

Reply:
Even though the base image includes them, verifying these versions explicitly ensures everything is set up as expected during deployment. Printing the versions doesn't cause any harm, and I've combined them into a single line for convenience. Having all the necessary information in one place can make troubleshooting easier when needed. I hope that's okay! 😊